ksqldb-serde/src/main/java/io/confluent/ksql/serde/protobuf/ProtobufSchemaTranslator.java

```diff
@@ -34,25 +34,46 @@
  * Translates between Connect and Protobuf Schema Registry schema types.
  */
 public class ProtobufSchemaTranslator implements ConnectSchemaTranslator {
+  private static final String DEFAULT_PREFIX_SCHEMA_NAME = "ConnectDefault";
 
-  private final ProtobufProperties properties;
   private final Map<String, Object> baseConfigs;
   private final Optional<String> fullNameSchema;
 
   private Map<String, Object> updatedConfigs;
   private ProtobufData protobufData;
 
   public ProtobufSchemaTranslator(final ProtobufProperties properties) {
-    this.properties = Objects.requireNonNull(properties, "properties");
+    Objects.requireNonNull(properties, "properties");
+
     this.baseConfigs = ImmutableMap.of(
-        WRAPPER_FOR_RAW_PRIMITIVES_CONFIG, properties.getUnwrapPrimitives());
-    this.fullNameSchema = Optional.ofNullable(
-        Strings.emptyToNull(Strings.nullToEmpty(properties.getFullSchemaName()).trim())
+        WRAPPER_FOR_RAW_PRIMITIVES_CONFIG, properties.getUnwrapPrimitives(),
+
+        // This flag is needed so that the schema translation in toConnectRow() adds the
+        // package name information to the row schema
+        ProtobufDataConfig.ENHANCED_PROTOBUF_SCHEMA_SUPPORT_CONFIG, "true"
     );
+    this.fullNameSchema = ignoreDefaultName(properties.getFullSchemaName());
     this.updatedConfigs = baseConfigs;
     this.protobufData = new ProtobufData(new ProtobufDataConfig(baseConfigs));
   }
 
+  private Optional<String> ignoreDefaultName(final String fullNameSchema) {
```

Comment:
Where does this schema name get populated? I'm wondering if we can avoid adding it, rather than adding it and then ignoring it if it's the default.

Reply:
It's like in 5 places. Two main sources are what this PR does when fetching the SR schema and create the schema translator based on the SR schema name. The other main one is when a user specifies the full schema name explicitly in the WITH properties. There are other places I don't know yet when they are called. So, to make it less prone to errors or miss other places, I just ignore it in the translator. It's just a property that the translator will ignore.